cuda_core/tests/test_module.py

```diff
@@ -15,13 +15,9 @@
 
 @pytest.mark.xfail(not can_load_generated_ptx(), reason="PTX version too new")
 def test_get_kernel():
-    kernel = """
-extern __device__ int B();
-extern __device__ int C(int a, int b);
-__global__ void A() { int result = C(B(), 1);}
-"""
+    kernel = """extern "C" __global__ void ABC() { }"""
     object_code = Program(kernel, "c++").compile("ptx", options=("-rdc=true",))
     assert object_code._handle is None
-    kernel = object_code.get_kernel("A")
+    kernel = object_code.get_kernel("ABC")
```

Comment:
I dunno why it passed when the CTK (NVRTC) version is higher than the driver version, but if we want to fetch a kernel by name, it has to have a C linkage (`extern "C"`).